user/tests.py

```diff
@@ -1,3 +1,237 @@
-from django.test import TestCase
+import json
+import bcrypt
+import jwt
 
-# Create your tests here.
+from django.test   import TestCase, Client
+from unittest.mock import MagicMock, patch
+
+from .models    import User
+from core.utils import (get_hashed_pw,
+                        checkpw,
+                        issue_token)
+
+
+class UserSignUpTest(TestCase):
+    def setUp(self):
+        self.URL = '/user/signup'
+        self.client = Client()
+        self.PASS_NAME = 'jaehoon'
+        self.PASS_EMAIL = 'jae@gmail.com'
+        self.PASS_PASSWORD = '12345678'
+
+        self.TEST_NAME = 'test'
+        self.TEST_EMAIL = 'test@gmail.com'
+        self.TEST_PASSWORD = '123456789'
+
+        self.user = User(
+            name=self.TEST_NAME,
+            email=self.TEST_EMAIL,
+            password=self.TEST_PASSWORD,
+        )
+        self.user.save()
+
+    def tearDown(self):
+        User.objects.all().delete()
+
+    def test_post_sign_up_success(self):
+        request = {
+            'name': self.PASS_NAME,
+            'email': self.PASS_EMAIL,
+            'password': self.PASS_PASSWORD
+        }
+
+        response = self.client.post(
+            self.URL, request, content_type='application/json')
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.json()['MESSAGE'], 'SUCCESS')
+
+    def test_post_sign_up_key_error(self):
+        requests = [
+            {'name': self.PASS_NAME},
+            {'email': self.PASS_EMAIL},
+            {'password': self.PASS_PASSWORD},
+            {
+                'name': self.PASS_NAME,
+                'password': self.PASS_PASSWORD
+            },
+            {
+                'name': self.PASS_NAME,
+                'email': self.PASS_EMAIL
+            },
+            {
+                'email': self.PASS_EMAIL,
+                'password': self.PASS_PASSWORD
+            }]
+
+        response = self.client.post(
+            self.URL, request, content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()['MESSAGE'], 'KEY_ERROR')
+
+    def test_post_sign_up_duplicate_info(self):
+        request = {
+            'name': self.TEST_NAME,
+            'email': self.TEST_EMAIL,
+            'password': self.TEST_PASSWORD
+        }
+
+        response = self.client.post(
+            self.URL, request, content_type='application/json')
+        self.assertEqual(response.status_code, 400)
```

Comment:
duplicate error code는 `409`입니다.
```suggestion
        self.assertEqual(response.status_code, 409)
```

Reply:
수정헀습니다